avocado/core/remote/runner.py

```diff
@@ -318,7 +318,9 @@ def setup(self):
 
     def tear_down(self):
         super(VMTestRunner, self).tear_down()
-        if self.job.args.vm_cleanup is True:
+        if (self.job.args.vm_cleanup is True and
```

Comment:
IMO, we should make sure the remote runner is called appropriately, that is, `tear_down` being called called before a `setup` is a bug in the caller context.  But, it doesn't hurt to have this extra check.